client/asset/dcr/dcr.go

```diff
@@ -2744,97 +2768,139 @@ func (dcr *ExchangeWallet) monitorBlocks(ctx context.Context) {
 // tipChange callback function is invoked and a goroutine is started to check
 // if any contracts in the findRedemptionQueue are redeemed in the new blocks.
 func (dcr *ExchangeWallet) checkForNewBlocks() {
-	ctx, cancel := context.WithTimeout(dcr.ctx, 2*time.Second)
-	defer cancel()
-	newTip, err := dcr.getBestBlock(ctx)
+	newTip, err := dcr.getBestBlock()
 	if err != nil {
 		go dcr.tipChange(fmt.Errorf("failed to get best block: %w", err))
 		return
 	}
 
-	// This method is called frequently. Don't hold write lock
-	// unless tip has changed.
-	dcr.tipMtx.RLock()
-	sameTip := dcr.currentTip.hash.IsEqual(newTip.hash)
-	dcr.tipMtx.RUnlock()
-	if sameTip {
+	cachedBestHash, cachedBestHeight := dcr.blockCache.Tip()
+	if newTip.hash.IsEqual(cachedBestHash) {
 		return
 	}
 
-	dcr.tipMtx.Lock()
-	defer dcr.tipMtx.Unlock()
+	newBestBlock, err := dcr.node.GetBlockVerbose(dcr.ctx, newTip.hash, false)
+	if err != nil {
+		go dcr.tipChange(fmt.Errorf("error retrieving block %s: %w", newTip.hash, translateRPCCancelErr(err)))
+		return
+	}
+
+	// Purge mainchain blocks if there was a reorg or this new block is NOT the
+	// successor of the currently cached best block. This MUST be done before
+	// adding the new best block to the cache and notifying consumers of the tip
+	// change.
+	var redemptionSearchStartBlock *block
+	err = func() error {
+		// If the new best block builds on the cached best block or the cache is empty,
+		// it's good to add.
+		if cachedBestHash == nil {
+			return nil // cache is empty
+		}
```

Comment:
In this case, `dcr.findRedemptionsInBlockRange` will start from the genesis block?